### PR TITLE
Task#17532 Ui for Option Bottom Sheet

### PR DESCRIPTION
**Pull request by kehaimat**
## [Pull request title]
Task#17532 Ui for Option Bottom Sheet
### [Redmine link]
https://edu-redmine.sun-asterisk.vn/issues/17532

### CODE REVIEW

File: app/src/main/java/vn/sunasterisk/music_70/ui/discover/DiscoverFragment.kt [hunk 1]
```diff
@@ -70,6 +71,7 @@ class DiscoverFragment : BaseFragment(), GenreContract.View {
                 )
             )
         }, {
+            fragmentManager?.let { fragman -> BottomSheetFragment.newInstance(it).show(fragman,BottomSheetFragment::class.java.name) }
```
Comment (phucblh-0954) [inline app/src/main/java/vn/sunasterisk/music_70/ui/discover/DiscoverFragment.kt]:
```suggestion
            fragmentManager?.let { fragment ->
                BottomSheetFragment.newInstance(it).show(fragment, BottomSheetFragment::class.java.name)
            }
```

File: app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+            }
+            R.id.textShare -> {
+                Toast.makeText(context, getString(R.string.string_share), Toast.LENGTH_LONG).show()
+                dismiss()
+            }
+
+            R.id.textInformation -> {
+                Toast.makeText(context, getString(R.string.string_information), Toast.LENGTH_LONG).show()
+                dismiss()
+            }
```
Comment (phucblh-0954) [inline app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt]:
```suggestion
            val noticeStringRes = when(v.id) {
                R.id.textDownload -> R.string.string_download
                R.id.textAddToFavourite -> R.string.string_add_to_favorite
                ...
            }
            Toast.makeText(context, noticeStringRes, Toast.LENGTH_LONG).show()
            dismiss()
```

File: app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        }
+    }
+
+    fun bindData(){
```
Comment (phucblh-0954) [inline app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt]: Ctrl + Alt + L

File: app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+    companion object {
+        fun newInstance(track: Track): BottomSheetFragment {
+            val args = Bundle().apply {
+                putParcelable(TrackAttributes.TRACK, track)
+            }
+            val fragment = BottomSheetFragment().apply {
+                arguments=args
+            }
+            return fragment
```
Comment (phucblh-0954) [inline app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt]:
```suggestion
            return BottomSheetFragment().apply {
                arguments = args
            }
```

**Review by kehaimat — COMMENTED** [review]
.

File: app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+
+class BottomSheetFragment : BottomSheetDialogFragment(), View.OnClickListener {
+    private lateinit var track: Track
+
+    override fun onClick(v: View) {
+        when (v.id) {
+            R.id.textDownload -> {
+                Toast.makeText(context, getString(R.string.string_download), Toast.LENGTH_LONG)
+                    .show()
+                dismiss()
```
Comment (datdt-1935) [inline app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt]: Có thể viết thành extension

File: app/src/main/res/drawable/ic_add_to_playlist.xml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+<vector xmlns:android="http://schemas.android.com/apk/res/android"
+    android:height="24dp"
+    android:width="24dp"
+    android:viewportWidth="24"
+    android:viewportHeight="24">
+    <path android:fillColor="#000" android:pathData="M17,3A2,2 0 0,1 19,5V21L12,18L5,21V5C5,3.89 5.9,3 7,3H17M11,7V9H9V11H11V13H13V11H15V9H13V7H11Z" />
+</vector>
```
Comment (datdt-1935) [inline app/src/main/res/drawable/ic_add_to_playlist.xml]: Enter end of file

File: app/src/main/java/vn/sunasterisk/music_70/ui/discover/DiscoverFragment.kt [hunk 7]
```diff
@@ -70,6 +71,8 @@ class DiscoverFragment : BaseFragment(), GenreContract.View {
                 )
             )
         }, {
+            fragmentManager?.let { fragment
+                -> BottomSheetFragment.newInstance(it).show(fragment,BottomSheetFragment::class.java.name) }
```
Comment (tuannm-1276) [inline app/src/main/java/vn/sunasterisk/music_70/ui/discover/DiscoverFragment.kt]: format code liên tục Ctrl + Alt + L & Ctrc + Alt + O

File: app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+import android.view.ViewGroup
+import android.widget.Toast
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment
+import kotlinx.android.synthetic.main.bottom_sheet_option.*
+import vn.sunasterisk.music_70.R
+import vn.sunasterisk.music_70.data.model.Track
+import vn.sunasterisk.music_70.data.remote.TrackAttributes
+import vn.sunasterisk.music_70.util.LoadImage
+
+
```
Comment (tuannm-1276) [inline app/src/main/java/vn/sunasterisk/music_70/ui/nowplaying/BottomSheetFragment.kt]: remove blank line

**Pull request merged**